variable/include/scipp/variable/variable.tcc

```diff
@@ -541,7 +541,10 @@ public:
   ElementArrayView<const value_type>
   valuesView(const Dimensions &dims, const Dim dim,
              const scipp::index begin) const override {
-    return {m_values, dims, dim, begin};
+    if (this->dims().contains(dim))
+      return {m_values, dims, dim, begin};
+    else
+      return {m_values, dims};
```

Comment:
Two questions:
- Is this the right place to fix this, or would it be better/possible to change the caller to use the overload a couple lines above?
- Would need to change this also for `variancesView`?